### PR TITLE
test: run tmt integration test on testing farm

### PR DESCRIPTION
**Pull request by henrywang**
This PR follows https://github.com/containers/bootc/pull/593 to run tmt integration test on testing farm. Two plan files has to be added:

- `build.fmf`: tmt plan file used to deploy testing farm runner and run tmt integration on the runner
- `build-tmt.fmf`: tmt test file used to make `.git` always present. Only `discover.fmf` setting is able to make tmt copy the git root instead of fmf root to the test directory

Testing farm:

- Command: `testing-farm request --plan /build --git-url https://github.com/henrywang/bootc.git --git-ref tf_run --compose Fedora-40 --arch x86_64`
- Result log (x86_64): https://artifacts.osci.redhat.com/testing-farm/6bd48984-4aaf-4dd1-9f85-cec5e1d11ce1/
- Duration: less than 10 minutes.

### CODE REVIEW

File: plans/build.fmf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  hardware:
+    cpu:
+      processors: ">= 2"
+    memory: ">= 6 GB"
+    virtualization:
+      is-supported: true
+prepare:
+  - how: install
+    package:
+      - cargo
```
Comment (cgwalters) [inline plans/build.fmf:11]: Can we change this to use `dnf builddep bootc` for the base requirements?

File: plans/integration.fmf [hunk 2]
```diff
@@ -4,8 +4,8 @@
 provision:
   how: virtual
   # Generated by `cargo xtask `
-  image: file://./target/testbootc-cloud.qcow2
+  image: file://./target/testvm/disk.qcow2
```
Comment (cgwalters) [inline plans/integration.fmf:7]: Oops, this was working for me since I had an old image there...ouch.

**Review by cgwalters — APPROVED** [review]

File: plans/build.fmf (hunk 1 above)
Comment (henrywang) [reply plans/build.fmf]: Sure. It's moved to `build-tmt.fmf`.

**Review by cgwalters — APPROVED** [review]
Thanks for working on this!

**Comment by cgwalters** [comment]
> Command: testing-farm request --plan /build --git-url https://github.com/henrywang/bootc.git --git-ref tf_run --compose Fedora-40 --arch x86_64

Cool. Just so I understand though, we haven't wired up this to PRs here, right?

**Comment by henrywang** [comment]
> Cool. Just so I understand though, we haven't wired up this to PRs here, right?

Right. PR https://github.com/containers/bootc/pull/615 finished tmt integration test.

**Pull request merged**